### PR TITLE
Fix up URL iteration

### PR DESCRIPTION
**Pull request by guybedford**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v8.x/docs/api/url.html#url_urlsearchparams_iterator

This fix then ensures that `global.URL` and `url.URL` interfaces are fully compatible and interchangeable.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -2129,17 +2129,17 @@ declare module "url" {
         constructor(init?: URLSearchParams | string | { [key: string]: string | string[] | undefined } | Iterable<string[]>);
```
Comment (Flarna) [inline types/node/index.d.ts]: Should we change the `Iterable<string[]>` here (and in the implements statement) also to `implements Iterable<[string, string]>`?

Comment (guybedford) [reply types/node/index.d.ts]: Great, point, I've included that as well.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -2126,20 +2126,20 @@ declare module "url" {
     }
 
     export class URLSearchParams implements Iterable<string[]> {
```
Comment (Flarna) [inline types/node/index.d.ts]: Sorry to comment again, but I think we should use also `Iterable<[string, string]>` here...

Comment (guybedford) [reply types/node/index.d.ts]: Thanks, added.

**Review by Flarna — APPROVED** [review]

**Comment by Flarna** [comment]
seems the last change cause a CI failure. I think this is because tsconfig targets es5 where arrays are no iterables. Not sure what the best solution is here and if it is acceptable to target es6.

**Comment by guybedford** [comment]
I tried adding the `string[][]` type as another option. Let me know if that works, or if you have any other suggestions?

File: types/node/index.d.ts [hunk 3]
```diff
@@ -2125,21 +2125,21 @@ declare module "url" {
         unicode?: boolean;
     }
 
-    export class URLSearchParams implements Iterable<string[]> {
-        constructor(init?: URLSearchParams | string | { [key: string]: string | string[] | undefined } | Iterable<string[]>);
+    export class URLSearchParams implements Iterable<[string, string]> {
+        constructor(init?: URLSearchParams | string | { [key: string]: string | string[] | undefined } | Iterable<[string, string]> | string[][]);
```
Comment (Flarna) [inline types/node/index.d.ts]: What about `Array<[string, string]>`? I think this matches best.

File: types/node/index.d.ts [hunk 4]
```diff
@@ -2125,21 +2125,21 @@ declare module "url" {
...
         has(name: string): boolean;
-        keys(): Iterator<string>;
+        keys(): IterableIterator<string>;
         set(name: string, value: string): void;
         sort(): void;
         toString(): string;
-        values(): Iterator<string>;
-        [Symbol.iterator](): Iterator<string[]>;
+        values(): IterableIterator<string>;
+        [Symbol.iterator](): IterableIterator<[string,string]>;
```
Comment (Flarna) [inline types/node/index.d.ts]: TSLint complains about a missing whitespace between , and string  :smirk:

**Comment by guybedford** [comment]
Sure, appreciated for the feedback.

**Pull request merged**